### ISSUE TITLE
Calling Java constructors are not considered interop

### ISSUE THREAD

**Issue by lassemaatta** [clj-kondo/clj-kondo#2774]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2026.01.19

**platform**

Native binary, Linux AMD64.

**editor**

Command line.

**problem**

Calling a Java class constructor is not considered as interop wrt. the `:warn-on-reflection` linter.

**repro**

Happy path, interop is detected due to `(Date/from bar)`.

```sh
echo '(ns foo (:import [java.util Date])) (def bar [])  (Date/from bar)' | clj-kondo --config '{:linters {:warn-on-reflection {:level :warning}}}' --repro  --debug  --lint -
[clj-kondo] Linting file: <stdin>
<stdin>:1:51: warning: Var *warn-on-reflection* is not set in this namespace.
linting took 3ms, errors: 0, warnings: 1
```

Sad path, interop is not detected even though we call `(Date. bar)`.

```sh
echo '(ns foo (:import [java.util Date])) (def bar [])  (Date. bar)' | clj-kondo --config '{:linters {:warn-on-reflection {:level :warning}}}' --repro  --debug  --lint - 
[clj-kondo] Linting file: <stdin>
linting took 6ms, errors: 0, warnings: 0
```

**config**

```clojure
{:linters {:warn-on-reflection {:level :warning}}}
```

**expected behavior**

```sh
echo '(ns foo (:import [java.util Date])) (def bar [])  (Date. bar)' | clj-kondo --config '{:linters {:warn-on-reflection {:level :warning}}}' --repro  --debug  --lint - 
[clj-kondo] Linting file: <stdin>
<stdin>:1:51: warning: Var *warn-on-reflection* is not set in this namespace.
linting took 3ms, errors: 0, warnings: 1
```
